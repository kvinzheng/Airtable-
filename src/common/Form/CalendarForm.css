.CalendarForm {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.4);
  z-index: 1;
}

.CalendarForm-start-time {
  outline: none;
  border: 1px solid #2f87be;
  border-radius: 4px;
  padding: 1rem 1rem;
  margin-bottom: 1.6rem;
  font-size: 17px;
  margin-right: 2rem;
  color: #757575;
}

.CalendarForm-required {
  border: 1px solid #c53c48;
}

.CalendarForm-input {
  outline: none;
  border: 1px solid #2f87be;
  border-radius: 15px;
  padding: 1rem 1rem;
  margin-bottom: 1.6rem;
  font-size: 15px;
  width: 100%;
  resize: none;
}

.CalendarForm-input-start,.CalendarForm-input-end{
  width:85%
}

.CalendarForm-input-datetime{
  display: inline-block;
  width: 50%;
}
.CalendarForm-input-datetime-label {
  width: 13%;
  display: inline-block;
  text-align: center;
}

.CalendarForm-content {
  background-color: #ffffff;
  padding: 15px 20px; 
  border-radius: 15px;
  width: 50%;
  min-width: 30rem;
  box-shadow: 1px 1px 5px #535050;
}

.CalendarForm-button span {
  display: block;
  padding: 12px 24px;
}

.CalendarForm-bottom {
  display: flex;
  justify-content: flex-end;
}

.CalendarForm-button {
  margin: 0 0.2rem 0 0.2rem;
  padding: 0;
  overflow: hidden;
  border-width: 0;
  outline: none;
  border-radius: 3px;
  background-color: #2f87be;
  color: #ecf0f1;
  font-weight: bold;
  transition: background-color 0.3s;
}

.CalendarForm-button:hover,
.CalendarForm-button:focus {
  background-color: #306180;
}

.CalendarForm-button-cancel {
  background-color: rgba(236, 240, 241, 0.3);
}




.CalendarForm-button-add {
  background-color: #2f87be;
}

.CalendarForm-button-add:hover,
.CalendarForm-button-add:focus {
  background-color: #2f87be;
  cursor: pointer;
}

.CalendarForm-button-del {
  background: none;
  border: 1px solid #555353;
  color: #555353;
  cursor: pointer;
}

.CalendarForm-button-cancel {
  background: none;
  color: #555353;
  cursor: pointer;
}

.CalendarForm-button-cancel:hover {
  background: none;
  background-color: transparent;  
  color: #555353;
  cursor: pointer;
}

.CalendarForm-button-del:hover,
.CalendarForm-button-del:focus {
  background-color: #97242c;
  color: white;
  border: 1px solid #97242c;
}

.CalendarForm-button:disabled {
  background: #a8a8a8;
  border: 1px solid #a8a8a8;
  cursor: default;
}

.CalendarForm-input-required {
  border: 1px solid #c53c48;
}

.CalendarForm-content-top {
  display: flex;
  justify-content: flex-end;
  margin: 10px 10px;
}