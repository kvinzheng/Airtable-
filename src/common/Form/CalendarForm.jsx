import React, { useState, useEffect } from "react";
import { makeISOdateString } from "../../helpers/dateFormat";
import "./CalendarForm.css";
import { setEvents, setEditableEvent } from "../../actions/events";
import { setCalendarForm } from "../../actions/form";
import { randomColor } from "randomcolor";
import { connect } from "react-redux";
import { AiOutlineClose } from "react-icons/ai";

export const CalendarForm = ({
  editableEvent,
  displayedDate,
  setEvents,
  setCalendarForm,
  events,
}) => {
  const [id, setId] = useState("");
  const [description, setDescription] = useState("");
  const [title, setTitle] = useState("");
  const [date, setDate] = useState(displayedDate);
  const [end, setEnd] = useState(displayedDate);
  useEffect(() => {
    if (editableEvent) {
      const { description, id, title, date, end } = editableEvent;
      setDescription(description);
      setId(id);
      setTitle(title);
      setDate(date);
      setEnd(end);
    }
  }, [editableEvent]);

  const inputValidation = () => {
    return title !== "" && !isNaN(Date.parse(date));
  };

  const handleEventsSet = (changed) => {
    let newEvents = [];
    if (changed.id === "") {
      changed.id += new Date();
      newEvents = [...events, changed];
    } else {
      newEvents = events.map((event) => {
        return event.id === changed.id ? { ...changed } : event;
      });
    }
    setEvents(newEvents);
    setCalendarForm(false);
  };

  const handleEventDelete = (id) => {
    const newEvents = events.filter((event) => event.id !== id);
    setCalendarForm(false);
    setEvents(newEvents);
  };

  const handleFormSubmit = (e) => {
    e.preventDefault();
    const value = {
      id,
      title,
      description,
      date,
      end,
      color: randomColor(),
    };
    handleEventsSet(value);
  };

  const newEvent = id === "" ? true : false;

  return (
    <div className="CalendarForm">
      <div className="CalendarForm-content">
        <div className="CalendarForm-content-top">
          <AiOutlineClose
            className="CalendarForm-button CalendarForm-button-cancel"
            size={25}
            onClick={() => setCalendarForm(false)}
          />
        </div>

        <form onSubmit={handleFormSubmit}>
          <div className="edit-form--top">
            <input
              className="CalendarForm-input CalendarForm-input-required"
              placeholder="Event name.."
              onChange={(e) => setTitle(e.target.value)}
              value={title}
            />
            <textarea
              className="CalendarForm-input"
              placeholder="Text Descriptions..."
              onChange={(e) => setDescription(e.target.value)}
              value={description}
            />
            <div className="CalendarForm-input-datetime">
              <label className="CalendarForm-input-datetime-label">From</label>
              <input
                className="CalendarForm-input  CalendarForm-input-required CalendarForm-input-start"
                type="datetime-local"
                onInput={(e) => setDate(new Date(e.target.value))}
                value={makeISOdateString(date)}
              />
            </div>

            <div className="CalendarForm-input-datetime">
              <label className="CalendarForm-input-datetime-label">To</label>
              <input
                className="CalendarForm-input  CalendarForm-input-required CalendarForm-input-end"
                type="datetime-local"
                onInput={(e) => setEnd(new Date(e.target.value))}
                value={makeISOdateString(end)}
              />
            </div>
          </div>

          <div className="CalendarForm-bottom">
            {!newEvent && (
              <button
                className="CalendarForm-button CalendarForm-button-del"
                type="button"
                onClick={() => handleEventDelete(id)}
              >
                <span>Delete</span>
              </button>
            )}
            <button
              className="CalendarForm-button CalendarForm-button-add"
              type="submit"
              disabled={!inputValidation()}
            >
              <span>{newEvent ? "Create" : "Edit"}</span>
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export const mapStateToProps = (state) => {
  return {
    events: state.events.data,
    editableEvent: state.events.editableEvent,
  };
};

export default connect(mapStateToProps, {
  setEvents,
  setCalendarForm,
  setEditableEvent,
})(CalendarForm);
