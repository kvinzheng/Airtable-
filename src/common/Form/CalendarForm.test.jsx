import React from "react";
import thunk from "redux-thunk";

import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { CalendarForm } from "./CalendarForm.jsx";
import { mapStateToProps } from "./CalendarForm";
import MockData from "../../helpers/testingData";

configure({ adapter: new Adapter() });

describe("CalendarForm Component", () => {
  it("should render with props passed in", () => {
    const today = new Date();
    const component = shallow(
      <CalendarForm
        events={MockData.events.data}
        form={MockData.form.CalendarForm}
        displayedDate={today}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("mapStateToProps", () => {
    const state = {
      events: { data: MockData.events.data, editableEvent: null },
    };
    const expected = {
      events: MockData.events.data,
      editableEvent: null,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });
});
