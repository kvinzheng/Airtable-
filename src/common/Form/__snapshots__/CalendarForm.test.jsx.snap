// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarForm Component should render with props passed in 1`] = `ShallowWrapper {}`;
